### ISSUE TITLE
key.png does not have green check

### ISSUE THREAD

**Issue by nicolas-raoul** [chrisjbillington/git_nautilus_icons#12]
Steps to reproduce:
1. Install per the README instructions
2. Kill then open Nautilus and browse the git_nautilus_icons folder itself
3. Expected: `key.png` has a green mark like all other files (I have not modified anything). Actually: No green mark.

![screenshot from 2019-02-18 11-11-37](https://user-images.githubusercontent.com/99590/52924048-2b74ba00-336e-11e9-8da6-ee62edae43be.png)

I tried twice.
Up-to-date Ubuntu 2018.04.1 LTS with Nautilus 3.26.4

**Comment by nicolas-raoul** [comment]
I set DEBUG to True and here is the log:
```
$ nautilus
sys:1: PyGIWarning: Nautilus was imported without specifying a version first. Use gi.require_version('Nautilus', '3.0') before import to ensure that the right version gets loaded.
Nautilus-Share-Message: 11:20:54.472: Called "net usershare info" but it failed: Failed to execute child process “net” (No such file or directory)
worker: git status loop: triggered
directory_status: /home/nico
repo status: /home/nico/git_nautilus_icons
parent: timeout
worker sending 0 processed files
parent: got response: STILL_WORKING
directory_status: /home/nico
parent: timeout
worker sending 0 processed files
parent: got response: STILL_WORKING
repo status: /home/nico/git_nautilus_icons
adding to ready set: /home/nico/git_nautilus_icons
worker: git status loop: triggered
parent: timeout
worker sending 1 processed files
parent: got response: ALL_DONE
adding icon for file: /home/nico/git_nautilus_icons
worker: git status loop: triggered
directory_status: /home/nico/git_nautilus_icons
repo status: /home/nico/git_nautilus_icons
parent: timeout
worker sending 0 processed files
parent: got response: STILL_WORKING
adding to ready set: /home/nico/git_nautilus_icons/key.png
adding to ready set: /home/nico/git_nautilus_icons/icons
adding to ready set: /home/nico/git_nautilus_icons/LICENSE
adding to ready set: /home/nico/git_nautilus_icons/git-nautilus-icons.py
adding to ready set: /home/nico/git_nautilus_icons/icon_testing_dir
adding to ready set: /home/nico/git_nautilus_icons/README.md
parent: timeout
worker sending 6 processed files
parent: got response: ALL_DONE
adding icon for file: /home/nico/git_nautilus_icons/git-nautilus-icons.py
adding icon for file: /home/nico/git_nautilus_icons/key.png
adding icon for file: /home/nico/git_nautilus_icons/README.md
adding icon for file: /home/nico/git_nautilus_icons/icon_testing_dir
adding icon for file: /home/nico/git_nautilus_icons/icons
adding icon for file: /home/nico/git_nautilus_icons/LICENSE
```

**Comment by chrisjbillington** [comment]
Thanks! I can reproduce this when I set nautilus to show file previews (I previously had it set to not show previews, in which case I don't observe the bug). Interestingly, I have another image file in the same repo (an untracked one) that gets an icon just fine when it is previewed:

![image](https://user-images.githubusercontent.com/1044087/52924472-f54a2280-32fa-11e9-8d08-af70010b851f.png)

I also get this warning from Nautilus:

```
(nautilus:16430): GdkPixbuf-CRITICAL **: 21:28:23.094: gdk_pixbuf_composite: assertion 'dest_x >= 0 && dest_x + dest_width <= dest->width' failed
```
Which sounds like maybe Nautilus is doing some incorrect pixel math, perhaps triggered by key.png's extreme aspect ratio.

I think I'll report this as a bug against Nautilus, since it looks like the extension is probably not doing anything wrong.

**Comment by nicolas-raoul** [comment]
Another one: 
![screenshot from 2019-02-18 11-43-51](https://user-images.githubusercontent.com/99590/52925005-8ad4c900-3372-11e9-9848-3a315696ec09.png)

**Comment by nicolas-raoul** [comment]
A few other ones. Unfortunately most of my files are images lol
![screenshot from 2019-02-18 11-48-46](https://user-images.githubusercontent.com/99590/52925206-64635d80-3373-11e9-8695-14694701b79c.png)
![screenshot from 2019-02-18 11-49-16](https://user-images.githubusercontent.com/99590/52925204-63cac700-3373-11e9-955a-7cbf5cf94301.png)
![screenshot from 2019-02-18 11-48-58](https://user-images.githubusercontent.com/99590/52925205-64635d80-3373-11e9-8130-78a0b28aaa94.png)

**Comment by chrisjbillington** [comment]
Thanks! This is more evidence that it affects image files with a high aspect ratio.

I've reported the bug upstream [here](https://gitlab.gnome.org/GNOME/nautilus/issues/912), and suspect there's not much that I can do in this extension unless it is fixed in Nautilus. Will wait to see what Nautilus devs say about the bug report in case I've misunderstood something.

**Comment by nicolas-raoul** [comment]
Nice bug report!
I guess you could create an GitHub label called `upstream` to mark issues like this one, where resolution requires upstream progress? Thanks!

**Comment by chrisjbillington** [comment]
[Patch submitted upstream](https://gitlab.gnome.org/GNOME/nautilus/merge_requests/423)

**Comment by chrisjbillington** [comment]
Patch was merged and released in nautilus 3.32.2
 
![image](https://user-images.githubusercontent.com/1044087/62551752-8c1f8280-b83a-11e9-8af4-aae29cf92a85.png)
